### PR TITLE
fix: Angular getting started guide - update client type

### PR DESCRIPTION
**Pull request by nadetastic**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by iartemiev — DISMISSED** [review]

File: src/pages/[platform]/start/getting-started/data-model/index.mdx [hunk 1]
```diff
@@ -440,7 +440,7 @@ import {
   ReactiveFormsModule,
   Validators
 } from '@angular/forms';
-import { generateClient } from 'aws-amplify/api';
+import { generateClient, Client } from 'aws-amplify/api';
```
Comment (renebrandel) [inline src/pages/[platform]/start/getting-started/data-model/index.mdx:443]:
```suggestion
import { generateClient, type Client } from 'aws-amplify/api';
```

File: src/pages/[platform]/start/getting-started/data-model/index.mdx [hunk 2]
```diff
@@ -514,7 +514,7 @@ import {
   Validators
 } from '@angular/forms';
 
-import { generateClient } from 'aws-amplify/api';
+import { generateClient, Client } from 'aws-amplify/api';
```
Comment (renebrandel) [inline src/pages/[platform]/start/getting-started/data-model/index.mdx:517]:
```suggestion
import { generateClient, type Client } from 'aws-amplify/api';
```

File: src/pages/[platform]/start/getting-started/data-model/index.mdx [hunk 3]
```diff
@@ -658,7 +658,7 @@ import {
...
```
Comment (renebrandel) [inline src/pages/[platform]/start/getting-started/data-model/index.mdx:661]:
```suggestion
import { generateClient, type Client } from 'aws-amplify/api';
```

**Review by renebrandel — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

File: src/pages/[platform]/start/getting-started/data-model/index.mdx [hunk 4]
```diff
@@ -440,7 +440,7 @@ import {
   ReactiveFormsModule,
   Validators
 } from '@angular/forms';
-import { generateClient } from 'aws-amplify/api';
+import { generateClient, type Client } from 'aws-amplify/api';
```
Comment (reesscot) [inline src/pages/[platform]/start/getting-started/data-model/index.mdx:443]: Is `type` necessary here? TS should know this is a type, and this reads weird.

```suggestion
import { generateClient, Client } from 'aws-amplify/api';
```

Comment (nadetastic) [reply src/pages/[platform]/start/getting-started/data-model/index.mdx:443]: I initially had it without but @renebrandel made this suggestion. I agree with it as it makes it clear to the reader it's a type being imported and not the client.

Comment (reesscot) [reply src/pages/[platform]/start/getting-started/data-model/index.mdx:443]: Just realized my feedback is the opposite of @renebrandel's. 🤣 I've usually only seen this like:
```
import type { Client } from 'aws-amplify/api';
```
Is it valid to put it inside the curly braces like this?

Comment (nadetastic) [reply src/pages/[platform]/start/getting-started/data-model/index.mdx:443]: Yeah it was introduced in typescript@4.5 looks like

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-5.html#type-modifiers-on-import-names

**Review by reesscot — APPROVED** [review]

**Pull request merged**